src/frontends/ir/tests/frontend_test_mmap.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "common_test_utils/file_utils.hpp"
+#include "frontend_test.hpp"
+#include "openvino/opsets/opset1.hpp"
+
+#ifndef __APPLE__  // TODO: add getVmRSSInKB() for Apple platform
+
+class IRFrontendMMapTestsAdvanced : public ::testing::Test, public IRFrontendTestsImpl {
+protected:
+    size_t binsize, REF_RSS;
+
+    void SetUp() override {
+        size_t SIZE_MB = 32;
+        size_t CONST_SIZE = SIZE_MB * 1024 * 1024 / sizeof(ov::element::f32);
+        auto parameter = std::make_shared<ov::opset1::Parameter>(ov::element::f32, ov::Shape{CONST_SIZE});
+        auto constant = std::make_shared<ov::opset1::Constant>(ov::element::f32,
+                                                               ov::Shape{CONST_SIZE},
+                                                               std::vector<float>(CONST_SIZE, 0));
+        auto add = std::make_shared<ov::opset1::Add>(parameter, constant);
+        auto result = std::make_shared<ov::opset1::Result>(add);
+        auto model = std::make_shared<ov::Model>(ov::NodeVector{result}, ov::ParameterVector{parameter});
+
+        auto filePrefix = ov::test::utils::generateTestFilePrefix();
+        xmlFileName = filePrefix + "_IrFrontendTestModel.xml";
+        binFileName = filePrefix + "_IrFrontendTestModel.bin";
+        ov::serialize(model, xmlFileName);
+        binsize = ov::test::utils::fileSize(binFileName) / 1024;
+
+        // In case of enabled `mmap` RAM should not increase more than 50% of .bin size
+        // Otherwise RAM should increase on at least 50% of .bin size
+        REF_RSS = binsize / 2;
+    }
+
+    void TearDown() override {
+        RemoveTemporalFiles();
+    }
+};
+
+TEST_F(IRFrontendMMapTestsAdvanced, core_enable_mmap_property) {
+    // Test checks that with  enabled `mmap` .bin file
+    // isn't read into RAM on `read_model` stage.
+    // Otherwise, with disabled `mmap` .bin file should
+    // be in RAM
+
+    auto test = [&](const bool& is_mmap) {
+        core.set_property(ov::enable_mmap(is_mmap));
+
+        auto rss_init = ov::test::utils::getVmRSSInKB();
+        auto model = core.read_model(xmlFileName);
+        auto rss_read = ov::test::utils::getVmRSSInKB();
+
+        bool is_weights_read = (rss_read - rss_init) > REF_RSS;
+        if (is_mmap == is_weights_read) {
+            std::cerr << "Test failed: mmap is " << (is_mmap ? "enabled" : "disabled") << ", but weights are "
+                      << (is_weights_read ? "read" : "not read") << " in RAM" << std::endl;
+            exit(1);
+        }
+        std::cerr << "Test passed" << std::endl;
```

Comment:
Why do you use `cerr` for notification message?
Moreover I think this message is duplicated in 67 line :)

Reply:
It's a design limitation of test in separate subprocess. Because `ASSERT_` and `EXPECT_` can't be handled, 2 processes communicate via stdout or stderr, and stdout isn't presented correctly in logs, so stderr is used. Duplicated message is a `regex` that means expected message that means test is correct

The reason why separate process used is to prevent any memory side effects from previous tests